### ISSUE TITLE
feat(probe): resolve a repo to its base model, and stop discarding the licence

### ISSUE THREAD

**Issue by yuriolive** [yuriolive/custom-ai#25]
Phase 2 of `docs/MARKETPLACE-DISCOVERY-PLAN.md`. Blocked by #24.

Resolve a Hugging Face repo to the base model it is a variant of, so the catalog can group
listings, and capture the licence in the same fetch. Both are properties of the base model,
not of the quant repo — a community re-quantization routinely carries `other` or omits the
licence entirely, and a permissive string on a quant repo does not relicense the weights
underneath it.

## The cascade (§2)

| # | Signal | Where | Action |
|---|---|---|---|
| 1 | `cardData.base_model` + `base_model_relation` | `GET /api/models/{id}` — `hf.ts` does not pass `?full=true` today | auto-link |
| 2 | `general.base_model.0.repo_url` | GGUF KV header — `parseGgufHeader` already returns the whole map, no new parsing | auto-link |
| 3 | Architecture fingerprint | every field is already probed and stored | **suggest only** |
| 4 | Normalized name | `deriveBaseName()` in `classify.ts` already strips `-GGUF`/`-i1`/quant suffixes | **suggest only** |

**Auto-link on 1 or 2 only.** A fine-tune has an identical architecture fingerprint to its
parent — same layers, same heads, same vocab — so any scheme that groups on architecture
alone merges `Qwen3-8B` with `SomeLab/Qwen3-8B-Uncensored` and presents the fine-tune's
output as the base model's. Signals 3 and 4 raise a suggestion the creator confirms:

```
This looks like  ● Qwen3 8B          (existing model, 2 other listings)
                 ○ A fine-tune of Qwen3 8B   → new model, Qwen3 8B as parent
                 ○ Something else            → new model, no parent
```

`base_model_relation` answers the same question without asking, which is why it ranks
first: `quantized` → link to the parent, `finetune`/`merge`/`adapter` → create a child.

Whatever fires is written to `custom_models.base_model_match`, so a wrong grouping is
explainable afterwards and a re-resolution pass can find every row grouped on a weak
signal.

## Licence capture

`lib/studio/server/model-card.ts:67` already parses the licence out of the model-card
frontmatter and **deliberately discards it**, so `license: apache-2.0` does not get
rendered as the model's description. Keep it instead of dropping it, and map it onto the
`commercial_hosting` enum (§5.1). An unresolved base model yields an `unknown` licence —
the two failures compose, and `unknown` never auto-publishes (that gate is #29).

## Scope

- `packages/hf-probe/src/hf.ts` — request `?full=true`, type `cardData`
- `packages/hf-probe/src/gguf.ts` — read `general.base_model.*`
- new `packages/hf-probe/src/identity.ts` — fingerprint + name scoring, pure, unit-tested
- `lib/studio/server/` — run the cascade in the deploy pipeline, persist the match
- `components/studio/` — the confirm step, only when no declared signal fired

## Done when

`npm run check` and `npm test` pass; `identity.ts` has node tests covering each signal and
the fine-tune-is-not-a-variant case; a GGUF repo with a declared base links without asking
the creator anything.
